05-Jason-Luo/wk07/3-wed/image-search-fix/js/search.js

```diff
@@ -0,0 +1,126 @@
+// console.log($.fn.jquery, _.VERSION);
+var pageNumber = 1;
+lastPage: false;
+
+const searchFlickr = function (keywords) {
+  if (lastPage) return;
+
+  console.log("Searching for", keywords);
+
+  const flickrURL = "https://api.flickr.com/services/rest";
+
+  $.getJSON(flickrURL, {
+    method: "flickr.photos.search",
+    api_key: "2f5ac274ecfac5a455f38745704ad084",
+    text: keywords,
+    format: "json",
+    page: pageNumber,
+    // pages: pageEnd,
+    nojsoncallback: 1, //don't bother looking this up: JSONP
+  })
+    .done(showImages)
+    .done(function (info) {
+      lastPage = info.photos.pages;
+      console.log(lastPage);
+      console.log(info);
+
+      if (info.photos.page >= info.photos.pages) {
+        lastPage = true;
+      }
+    });
+  pageNumber++;
+};
+
+const showImages = function (results) {
+  //   console.table(results.photos.photo);
+  //   _(results.photos.photo).each(function (photo) {
+  // console.log(photo);
+  // generate a URL
+  // display the image
+  //   });
+  const urls = _(results.photos.photo).map(generateURL);
+  //   console.log(urls);
+  _(urls).each(function (url) {
+    // const $img = $("<img>").attr("src", url);
+    // const $img = $(`<img src=${url}>`);
+    const $img = $("<img>", { src: url });
+    $img.appendTo("#images");
+  });
+
+  // if (results.length === 0) {
+  //   // pause infiniate scroll if empty
+  //   $("#search").infinitescroll("pause");
+  // }
+};
+
+const generateURL = function (p) {
+  return [
+    "http://farm",
+    p.farm,
+    ".static.flickr.com/",
+    p.server,
+    "/",
+    p.id,
+    "_",
+    p.secret,
+    "_q.jpg", // change 'q' to something else for different sizes (see documentation)
+  ].join("");
+};
+
+$(document).ready(function () {
+  $("#search").on("submit", function (event) {
```

Comment:
```suggestion
  $("#search").submit(function (event) {
```
A lot of events have dedicated shorthand methods to add listeners like this.